sky/server/requests/requests.py

```diff
@@ -703,3 +712,51 @@ def set_request_cancelled(request_id: str) -> None:
     with update_request(request_id) as request_task:
         assert request_task is not None, request_id
         request_task.status = RequestStatus.CANCELLED
+
+
+@init_db
+def clean_finished_requests_with_retention(retention_seconds: int) -> None:
+    """Clean up finished requests older than the retention period.
+
+    This function removes old finished requests (SUCCEEDED, FAILED, CANCELLED)
+    from the database and cleans up their associated log files.
+
+    Args:
+        retention_seconds: Requests older than this many seconds will be
+            deleted.
+    """
+    finished_statuses = [
+        RequestStatus.SUCCEEDED, RequestStatus.FAILED, RequestStatus.CANCELLED
+    ]
+    reqs = get_request_tasks(status=finished_statuses,
+                             created_before=time.time() - retention_seconds)
+
+    for req in reqs:
+        req.log_path.unlink(missing_ok=True)
+
+    id_list_str = ','.join(repr(req.request_id) for req in reqs)
+    assert _DB is not None
+    with _DB.conn:
+        cursor = _DB.conn.cursor()
+        cursor.execute(
+            f'DELETE FROM {REQUEST_TABLE} WHERE request_id IN ({id_list_str})')
+
+    logger.info(f'Cleaned up {len(reqs)} finished requests '
+                f'older than {retention_seconds} seconds')
+
+
+def requests_gc_daemon():
+    """Garbage collect finished requests periodically."""
+    retention_seconds = skypilot_config.get_nested(
+        ['api_server', 'requests_gc_retention_seconds'],
+        DEFAULT_REQUESTS_GC_RETENTION_SECONDS)
```

Comment:
with this, if a user change the config in dashboard to a smaller value, it won't take effect in the current API server? Wondering if we need a way to reload the config.